misc/services/camel/de-metas-camel-externalsystems/core/src/main/java/de/metas/camel/externalsystems/core/to_mf/v2/ProductRouteBuilder.java

```diff
@@ -70,6 +75,29 @@ public void configure() throws Exception
 				.setHeader(CoreConstants.AUTHORIZATION, simple(CoreConstants.AUTHORIZATION_TOKEN))
 				.setHeader(Exchange.HTTP_METHOD, constant(HttpEndpointBuilderFactory.HttpMethods.GET))
 				.toD("http://{{metasfresh.products.v2.api.uri}}?${header.queryParams}");
+
+		from(direct(MF_UPSERT_PRODUCT_V2_CAMEL_URI))
+				.routeId(MF_UPSERT_PRODUCT_V2_CAMEL_URI)
+				.streamCaching()
+				.process(exchange -> {
+					final var lookupRequest = exchange.getIn().getBody();
+					if (!(lookupRequest instanceof ProductUpsertCamelRequest))
+					{
+						throw new RuntimeCamelException("The route " + MF_UPSERT_PRODUCT_V2_CAMEL_URI + " requires the body to be instanceof ProductUpsertCamelRequest V2."
+																+ " However, it is " + (lookupRequest == null ? "null" : lookupRequest.getClass().getName()));
+					}
+
+					exchange.getIn().setHeader(HEADER_ORG_CODE, ((ProductUpsertCamelRequest)lookupRequest).getOrgCode());
+					final var jsonRequestProductUpsert = ((ProductUpsertCamelRequest)lookupRequest).getJsonRequestProductUpsert();
+
+					log.info("Product upsert route invoked with " + jsonRequestProductUpsert.getRequestItems().size() + " requestItems");
+					exchange.getIn().setBody(jsonRequestProductUpsert);
+				})
+				.marshal(CamelRouteHelper.setupJacksonDataFormatFor(getContext(), JsonRequestProductUpsert.class))
+				.removeHeaders("CamelHttp*")
+				.setHeader(CoreConstants.AUTHORIZATION, simple(CoreConstants.AUTHORIZATION_TOKEN))
+				.setHeader(Exchange.HTTP_METHOD, constant(HttpEndpointBuilderFactory.HttpMethods.PUT))
+				.toD("http://{{metasfresh.upsert-product-v2.api.uri}}/${header." + HEADER_ORG_CODE + "}");
```

Comment:
I just thought, why do we have the hardcoded `http://` ? wouldn't it be better to include the protocol in the respective URI as well?
Because in future we might want to run camel a bit more remote and then we'd might need to use `https://` or whatever..
Not urgent for now, but I think it will be part of one of the next issues..